apps/emqx_ds_shared_sub/src/emqx_ds_shared_sub_store.erl

```diff
@@ -263,67 +284,87 @@ mk_leader_topic(GroupName) ->
 %% in RPCs between different nodes of potentially different releases.
 -type t() :: #{
     %% General.
-    group := group(),
+    id := id(),
     %% Spaces and variables: most up-to-date in-memory state.
+    properties := #{
+        %% TODO: Efficient encoding.
+        topic => emqx_types:topic(),
+        start_time => emqx_message:timestamp(),
+        created_at => emqx_message:timestamp()
+    },
     stream := #{emqx_ds:stream() => stream_state()},
-    start_time => _SubsriptionStartTime :: emqx_message:timestamp(),
     rank_progress => _RankProgress,
-    %% Internal _sequence number_ that tracks every change.
+    %% Internal _sequence numbers_ that tracks every change.
     seqnum := integer(),
     %% Mapping between complex keys and seqnums.
     seqmap := #{space_key() => _SeqNum :: integer()},
+    %% Last committed sequence number.
+    committed := integer(),
     %% Stage: uncommitted changes.
-    stage := #{space_key() | var_name() => _Value},
-    dirty => true
+    stage := #{space_key() | var_name() => _Value}
 }.
 
 -type stream_state() :: #{
     progress => emqx_persistent_session_ds_shared_subs:progress(),
     rank => emqx_ds:stream_rank()
 }.
 
--spec init(group()) -> t().
-init(Group) ->
-    %% NOTE: Empty store is dirty because rootset needs to be persisted.
-    mark_dirty(mk_store(Group)).
+-spec init(id()) -> t().
+init(ID) ->
+    %% NOTE: Empty store is impicitly dirty because rootset needs to be persisted.
+    mk_store(ID).
+
+-spec open(id()) -> t() | false.
+open(ID) ->
+    case open_rootset(ID) of
+        Rootset = #{} ->
+            slurp_store(Rootset, mk_store(ID));
+        Otherwise ->
+            Otherwise
+    end.
 
--spec open(group()) -> t() | false.
-open(Group) ->
-    open_store(mk_store(Group)).
+-spec exists(id()) -> boolean().
+exists(ID) ->
+    open_rootset(ID) =/= false.
 
-mk_store(Group) ->
+mk_store(ID) ->
     #{
-        group => Group,
+        id => ID,
         stream => #{},
+        properties => #{},
         seqnum => 0,
         seqmap => #{},
+        committed => 0,
         stage => #{}
     }.
 
-open_store(Store = #{group := Group}) ->
-    ReadRootset = mk_read_root_batch(Group),
+open_rootset(ID) ->
```

Comment:
If we implement synchronous version of `get_streams`, will this simplify to a simple read operation?